src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSTempPECompiler.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS
+{
+    [Export(typeof(ITempPECompiler))]
+    internal class CPSTempPECompiler : ITempPECompiler
+    {
+        private readonly VisualStudioWorkspace _workspace;
+
+        [ImportingConstructor]
+        public CPSTempPECompiler(VisualStudioWorkspace workspace)
+        {
+            _workspace = workspace;
+        }
+
+        public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string outputFileName, HashSet<string> filesToInclude, CancellationToken cancellationToken)
```

Comment:
Oh, this should probably take an ISet instead of a HashSet; if you have a different set under the covers we still don't want to spent garbage converting it.